editor/plugins/script_text_editor.cpp

```diff
@@ -1147,6 +1148,32 @@ void ScriptTextEditor::_edit_option(int p_op) {
 
 			_convert_case(CodeTextEditor::CAPITALIZE);
 		} break;
+		case EDIT_EVALUATE: {
+
+			Expression expression;
+			Vector<String> lines = code_editor->get_text_edit()->get_selection_text().split("\n");
+			PoolStringArray results;
+
+			for (int i = 0; i < lines.size(); i++) {
+				String line = lines[i];
+				String whitespace = line.substr(0, line.size() - line.strip_edges(true, false).size()); //extract the whitespace at the beginning
+
+				if (expression.parse(line) == OK) {
+					Variant result = expression.execute(Array(), Variant(), false);
+					if (expression.get_error_text() == "") {
+						results.append(whitespace + (String)result);
+					} else {
+						results.append(line);
+					}
+				} else {
+					results.append(line);
+				}
+			}
+
+			code_editor->get_text_edit()->begin_complex_operation(); //prevents creating a two-step undo
```

Comment:
Will it actually create a two-step undo though?

Reply:
Yes, I added the fix because it was happening.